nncf/common/pruning/statistics.py

```diff
@@ -0,0 +1,164 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import List, Dict, Any
+
+from nncf.api.compression import Statistics
+from nncf.common.utils.helpers import create_table
+
+
+class PrunedLayerSummary:
+    """
+    Contains information about the pruned layer.
+    """
+
+    def __init__(self,
+                 name: str,
+                 weight_shape: List[int],
+                 mask_shape: List[int],
+                 weight_pruning_level: float,
+                 mask_pruning_level: float,
+                 filter_pruning_level: float):
+        """
+        Initializes a summary about the pruned layer.
+
+        :param name: Layer's name.
+        :param weight_shape: Weight's shape.
+        :param mask_shape: Mask's shape.
+        :param weight_pruning_level: TODO
+        :param mask_pruning_level: TODO
+        :param filter_pruning_level: TODO
+        """
+        self.name = name
+        self.weight_shape = weight_shape
+        self.mask_shape = mask_shape
+        self.weight_pruning_level = weight_pruning_level
+        self.mask_pruning_level = mask_pruning_level
+        self.filter_pruning_level = filter_pruning_level
+
+    def as_dict(self) -> Dict[str, Any]:
+        """
+        Returns a representation of the pruned layer's summary as built-in data types.
+
+        :return: A representation of the sparsified layer's summary as built-in data types.
+        """
+        summary = {
+            'name': self.name,
+            'weight_shape': self.weight_shape,
+            'mask_shape': self.mask_shape,
+            'weight_pruning_level': self.weight_pruning_level,
+            'mask_pruning_level': self.mask_pruning_level,
+            'filter_pruning_level': self.filter_pruning_level,
+        }
+        return summary
```

Comment:
Note that this is essentially a copy-and-paste of the corresponding `self.__dict__`. If you decide to take this approach, then perhaps it would be easier to write a common `Statistics` function that iterates over `self.__dict__`, puts built-in types (int, double and str) into the retval dict as-is, puts the result of `.as_dict` for `Statistics` entries and ignores the rest (if encountered, spawn a debug message with a warning).

Reply:
The `as_dict()` method was removed after refactoring. I implemented a helper function that prepares statistics for the tensor board.